src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -20,29 +25,68 @@
     private Product product1;
     @Mock
     private Product product2;
+    @Mock
+    private Product product3;
+    @Mock
+    private Product product4;
 
-    private ArrayListProductDao productDao = new ArrayListProductDao();
+    @InjectMocks
+    private ArrayListProductDao productDao = ArrayListProductDao.getInstance();
 
     @Before
     public void setup() {
-        when(product1.getId()).thenReturn(1L);
-        when(product2.getId()).thenReturn(null);
-        when(product2.getStock()).thenReturn(1);
-        when(product2.getPrice()).thenReturn(new BigDecimal(100));
+        changeProduct(product1, 1L, 1, "Samsung Galaxy S", new BigDecimal(100));
+        changeProduct(product2, 2L, 1, "Apple iPhone", new BigDecimal(200));
+        changeProduct(product3, 3L, 0, "Siemens C56", new BigDecimal(300));
+        changeProduct(product4, null, 1, "Palm Pixi", new BigDecimal(200));
+        productDao.setProducts(new ArrayList<>(Arrays.asList(product1, product2, product3)));
+    }
+
+    public void changeProduct(Product product, Long id, int stock, String description, BigDecimal price) {
+        when(product.getId()).thenReturn(id);
+        when(product.getStock()).thenReturn(stock);
+        when(product.getPrice()).thenReturn(price);
+        when(product.getDescription()).thenReturn(description);
     }
 
     @Test
-    public void testFindProductsResults() {
-        assertFalse(productDao.findProducts().isEmpty());
+    public void testFindProducts() {
+        assertEquals(2, productDao.findProducts().size());
```

Comment:
Please do not test this method by size now. Verify that the expected products are returned in it